### PR TITLE
Spring  @Async plugin support

### PR DESCRIPTION
**Pull request by zhaoyuguang**
Please answer these questions before submitting pull request

- Why submit this pull request?
- [ ] Bug fix
- [x] New feature provided
- [ ] Improve performance

- Related issues
https://github.com/apache/skywalking/issues/2806

### CODE REVIEW

**Comment by wu-sheng** [comment]
Integration test cases are required.

File: apm-sniffer/apm-sdk-plugin/spring-plugins/async-annotation-5.x-plugin/pom.xml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    <spring-aop.version>5.1.4.RELEASE</spring-aop.version>
+</properties>
+
+<dependencies>
+    <dependency>
+        <groupId>org.springframework</groupId>
+        <artifactId>spring-aop</artifactId>
+        <version>${spring-aop.version}</version>
+    </dependency>
+</dependencies>
```
Comment (wu-sheng) [inline apm-sniffer/apm-sdk-plugin/spring-plugins/async-annotation-5.x-plugin/pom.xml]: The pom format looks like very strange.

Comment (zhaoyuguang) [reply apm-sniffer/apm-sdk-plugin/spring-plugins/async-annotation-5.x-plugin/pom.xml]: Ok

**Comment by zhaoyuguang** [comment]
> Integration test cases are required.

Ok, I will support it soon

**Comment by asfgit** [comment]
FAILURE

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/skywalking-e2e/9/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/skywalking-ci-it/64/

**Comment by zhaoyuguang** [comment]
@IanCao  Please do testcase

**Comment by wu-sheng** [comment]
@zhaoyuguang @IanCao What is the status of this PR?

**Comment by wu-sheng** [comment]
@zhaoyuguang You breaks the e2e by compiling error. Please recheck, look like your check style file is missing?

cc @kezhenxu94

**Comment by wu-sheng** [comment]
Look like because version updated since 6.2.0 release. Please fix.

**Comment by zhaoyuguang** [comment]
> Look like because version updated since 6.2.0 release. Please fix.

Ok, I will fix it

**Comment by zhaoyuguang** [comment]
> @zhaoyuguang @IanCao What is the status of this PR?

I have completed the integration test and it looks good, is it possible to do CodeReview?

File: apm-sniffer/apm-sdk-plugin/spring-plugins/async-annotation-plugin/pom.xml [hunk 2]
```diff
@@ -21,7 +21,7 @@
     <parent>
         <artifactId>spring-plugins</artifactId>
         <groupId>org.apache.skywalking</groupId>
-        <version>6.2.0-SNAPSHOT</version>
+        <version>6.3.0-SNAPSHOT</version>
```
Comment (kezhenxu94) [inline apm-sniffer/apm-sdk-plugin/spring-plugins/async-annotation-plugin/pom.xml]: Why not use `${project.version}`?

Comment (wu-sheng) [reply apm-sniffer/apm-sdk-plugin/spring-plugins/async-annotation-plugin/pom.xml]: I think you can't set this in parent paragraph.

**Comment by wu-sheng** [comment]
@zhaoyuguang Have you finished your review?

File: apm-sniffer/apm-sdk-plugin/spring-plugins/async-annotation-plugin/pom.xml (hunk 2 above)
Comment (kezhenxu94) [reply apm-sniffer/apm-sdk-plugin/spring-plugins/async-annotation-plugin/pom.xml]: > I think you can't set this in parent paragraph.

Oh yes, forgive me

File: apm-sniffer/apm-sdk-plugin/spring-plugins/async-annotation-plugin/src/main/java/org/apache/skywalking/apm/plugin/spring/async/SWCallable.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+import org.apache.skywalking.apm.network.trace.component.ComponentsDefine;
+
+import java.util.concurrent.Callable;
+
+/**
+ * @author zhaoyuguang
+ */
+public class SWCallable<V> implements Callable<V> {
+
+    private static String OPERATION_NAME = "SpringAsync";
```
Comment (IanCao) [inline apm-sniffer/apm-sdk-plugin/spring-plugins/async-annotation-plugin/src/main/java/org/apache/skywalking/apm/plugin/spring/async/SWCallable.java]: make it final is better

Comment (zhaoyuguang) [reply apm-sniffer/apm-sdk-plugin/spring-plugins/async-annotation-plugin/src/main/java/org/apache/skywalking/apm/plugin/spring/async/SWCallable.java]: ok

File: docs/en/setup/service-agent/java-agent/Supported-list.md [hunk 4]
```diff
@@ -73,6 +73,9 @@
 * Vert.x Ecosystem
   * Vert.x Eventbus 3.2+
   * Vert.x Web 3.x
+* Multi-threaded Framework
```
Comment (wu-sheng) [inline docs/en/setup/service-agent/java-agent/Supported-list.md]: Thread Schedule Framework

I think we could add more threadpool plugin in this catalog.

Comment (zhaoyuguang) [reply docs/en/setup/service-agent/java-agent/Supported-list.md]: ok

**Comment by wu-sheng** [comment]
This plugin breads e2e?

**Comment by wu-sheng** [comment]
@kezhenxu94 Please help on the first e2e check failure. :) Interesting

**Comment by wu-sheng** [comment]
I click the re-run to see what happens.

**Comment by wu-sheng** [comment]
@asfgit rerun the e2e tests please

**Pull request merged**